include/zephyr/storage/storage_area/storage_area_eeprom.h

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2024 Laczen
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Public API for storage area eeprom subsystem
+ */
+
+#ifndef ZEPHYR_INCLUDE_STORAGE_STORAGE_AREA_EEPROM_H_
+#define ZEPHYR_INCLUDE_STORAGE_STORAGE_AREA_EEPROM_H_
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/storage/storage_area/storage_area.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Storage_area_eeprom interface
+ * @defgroup storage_area_eeprom_interface Storage_area_eeprom interface
+ * @ingroup storage_apis
+ * @{
+ */
+
+struct storage_area_eeprom {
+	const struct storage_area area;
+	const struct device *dev;
+	const size_t start;
+};
+
+extern const struct storage_area_api storage_area_eeprom_api;
+
+#define eeprom_storage_area(_dev, _start, _ws, _es, _size, _props)              \
```

Comment:
I would prefer to pass a name argument here and verify parameters with build asserts:

```c
#define STORAGE_AREA_EEPROM_DEFINE(_name, _dev, _start, _ws, _es, _size, _props)    \
	BUILD_ASSERT(_ws != 0, "Invalid EEPROM write size");                        \
	BUILD_ASSERT((_ws & (_ws - 1) == 0, "Invalid EEPROM write size");           \
	BUILD_ASSERT((_es % _ws) == 0, "Invalid EEPROM erase size");                \
	BUILD_ASSERT((_size % _ws) == 0, "Invalid EEPROM size");                    \
	const struct storage_area_eeprom _name = {                                  \
		.area =                                                             \
			{                                                           \
				.api = &storage_area_eeprom_api,                    \
				.write_size = _ws,                                  \
				.erase_size = _es,                                  \
				.erase_blocks = _size / _es,                        \
				.props = _props | SA_PROP_FOVRWRITE,                \
			},                                                          \
		.dev = _dev, .start = _start,                                       \
		.dev = _dev, .start = _start,                                       \
	}

```

Reply:
Fixed